nexussdk/project.py

```diff
@@ -2,26 +2,165 @@
 from . utils.http import http_get
 from . utils.http import is_response_valid
 from . utils.http import http_put
-from . utils.http import http_post
 from . utils.http import http_delete
-from . utils.tools import copy_this_into_that
 import urllib.parse
 
 
-def get(org_label, project_label):
-    print("Not implemented yet")
+def fetch(org_label, project_label, rev=None):
+    """
+    Fetch a project and all its details.
+    Note: This does not give the list of resources. To get that, use the `resource` package.
 
-def create(org_label, project_label):
-    print("Not implemented yet")
+    :param org_label: The label of the organization that contains the project to be fetched
+    :param project_label: label of a the project to fetch
+    :param rev: OPTIONAL The specific revision of the wanted project. If not provided, will get the last.
+    :return: All the details of this project, as a dictionary
+    """
 
-def update(org_label, project_label, previous_rev):
-    print("Not implemented yet")
+    org_label = urllib.parse.quote_plus(org_label)
+    project_label = urllib.parse.quote_plus(project_label)
+    url = "/projects/" + org_label + "/" + project_label
 
-def tag(org_label, project_label, previous_rev, tag):
-    print("Not implemented yet")
+    if rev is not None:
+        url = url + "?rev=" + str(rev)
 
-def list(org_label):
-    print("Not implemented yet")
+    response_raw = http_get(url)
 
-def deprecate(org_label, project_label):
-    print("Not implemented yet")
+    if not is_response_valid(response_raw):
+        raise Exception("Invalid http request for " + response_raw.url +
+                        " (Status " + str(response_raw.status_code) + ")" + "\n" +
+                        response_raw.text)
+
+    response_obj = json.loads(response_raw.text)
+    return response_obj
+
+
+def create(org_label, project_label, data=None):
```

Comment:
I would call this argument config instead of data.

Reply:
true, it's more a config than actual data. Changing the code.